desktop_version/src/main.cpp

```diff
@@ -153,6 +153,8 @@ int main(int argc, char *argv[])
         SDL_INIT_JOYSTICK |
         SDL_INIT_GAMECONTROLLER
     );
+    // WTF? Why is text input active from program start?!
+    SDL_StopTextInput();
```

Comment:
This is invalid behavior - SDL makes it perfectly clear that text input is enabled on targets where a keyboard is always present. It's only disabled for devices that have to use on-screen keyboards, where you would have to explicitly start it to know for sure that you will receive text events.

Reply:
Okay, but if I don't do this, then SDL will tell the game text input is being actively taken when it isn't, leading to being unable to do anything in the editor. Is there another way to work around this that I'm not aware of?

Reply:
`SDL_IsTextInputActive()` should be accurate at this point in the program.

Reply:
Well, that's the thing. When I tested it, it was not. If I don't do this, then going to the editor results in my keypresses doing nothing until I press Esc.

Reply:
I find this hard to believe, here is the implementation:

https://hg.libsdl.org/SDL/file/369fea168258/src/video/SDL_video.c#l3836

SDL_InitEvents will start the event state as disabled...

https://hg.libsdl.org/SDL/file/369fea168258/src/events/SDL_events.c#l419

... but VideoInit will enable it based on the backend's screen keyboard support:

https://hg.libsdl.org/SDL/file/369fea168258/src/video/SDL_video.c#l556

We know the event state will be disabled before the above line since the video subsystem requires that the event subsystem be initialized first:

https://hg.libsdl.org/SDL/file/369fea168258/src/video/SDL_video.c#l479

You will want to make absolutely sure that you are disabling it for the right case, as Start/Stop do a lot more than just enable/disable the event type.

Reply:
I don't really see any part of your explanation where it's clear that the event state is disabled at the end of `SDL_Init()`. Especially since line 556 of `SDL_video.c` (in `SDL_VideoInit()`) does `SDL_StartTextInput()` if `SDL_HasScreenKeyboardSupport()` is false.

Indeed, if I open the game up in GDB and set a breakpoint on the first line of `SDL_StartTextInput()`, `SDL_video.c:3757` (it seems like directly setting a breakpoint on the function name doesn't work, I assume this is because of inlining), that's exactly what happens. `SDL_VideoInit()` ends up calling `SDL_StartTextInput()`.

Reply:
Correct, that is why I expect SDL_IsTextInputActive() to return 1 and not 0. You need to check to see if it's active before stopping input, otherwise you enter potentially undefined behavior. If it's returning 0, you need to find out why this is the case.

Reply:
Fair enough, but then I'm confused on how much we actually disagreed in the first place.

Whatever. I can add the check so that it only disables it if it's enabled, but I'm not sure if you want me to do more than that (judging by the sentence "If it's returning 0, you need to find out why this is the case").